nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/pcap/PCAP.java

```diff
@@ -0,0 +1,147 @@
+        headerBuffer.put(this.readIntToNBytes(this.hdr().thiszone(), 4, false));
+        headerBuffer.put(this.readLongToNBytes(this.hdr().sigfigs(), 4, true));
+        headerBuffer.put(this.readLongToNBytes(this.hdr().snaplen(), 4, true));
+        headerBuffer.put(this.readLongToNBytes(this.hdr().network(), 4, true));
+
+        List<byte[]> packetByteArrays = new ArrayList<>();
+
+        int packetBufferSize = 0;
+
+        for (Packet currentPacket : packets) {
+            int currentPacketTotalLength = Packet.PACKET_HEADER_LENGTH + currentPacket.rawBody().length;
+
+            ByteBuffer currentPacketBytes = ByteBuffer.allocate(currentPacketTotalLength);
+            currentPacketBytes.put(readLongToNBytes(currentPacket.tsSec(), 4, false));
+            currentPacketBytes.put(readLongToNBytes(currentPacket.tsUsec(), 4, false));
+            currentPacketBytes.put(readLongToNBytes(currentPacket.inclLen(), 4, false));
+            currentPacketBytes.put(readLongToNBytes(currentPacket.origLen(), 4, false));
+            currentPacketBytes.put(currentPacket.rawBody());
+
+            packetByteArrays.add(currentPacketBytes.array());
+            packetBufferSize += currentPacketTotalLength;
+        }
+
+        ByteBuffer packetBuffer = ByteBuffer.allocate(packetBufferSize);
+        packetBuffer.order(ByteOrder.LITTLE_ENDIAN);
+
+        for (byte[] packetByteArray : packetByteArrays) {
+            packetBuffer.put(packetByteArray);
+        }
+
+        ByteBuffer allBytes = ByteBuffer.allocate(headerBufferSize + packetBufferSize);
+        allBytes.order(ByteOrder.LITTLE_ENDIAN);
+
+        allBytes.put(headerBuffer.array());
+        allBytes.put(packetBuffer.array());
+
+        return allBytes.array();
+    }
+
+    private byte[] readIntToNBytes(int input, int numberOfBytes, boolean isSigned) {
+        byte[] output = new byte[numberOfBytes];
+        output[0] = (byte) (input & 0xff);
+        for (int loop = 1; loop < numberOfBytes; loop++) {
+            if (isSigned) {
+                output[loop] = (byte) (input >> (8 * loop));
+            } else {
+                output[loop] = (byte) (input >>> (8 * loop));
+            }
+        }
+        return output;
+    }
+
+    private byte[] readLongToNBytes(long input, int numberOfBytes, boolean isSigned) {
+        byte[] output = new byte[numberOfBytes];
+        output[0] = (byte) (input & 0xff);
+        for (int loop = 1; loop < numberOfBytes; loop++) {
+            if (isSigned) {
+                output[loop] = (byte) (input >> (8 * loop));
+            } else {
+                output[loop] = (byte) (input >>> (8 * loop));
+            }
+        }
+        return output;
+    }
+
+    private void read() {
+        this.hdr = new Header(this.io);
+        this.packets = new ArrayList<>();
+        {
+            while (!this.io.isEof()) {
+                this.packets.add(new Packet(this.io, this));
+            }
+        }
```

Comment:
Sorry for not requesting this earlier as I had seen this but forgotten about it. This block seems unnecessary.
```suggestion
```suggestion
        this.hdr = new Header(this.io);
        this.packets = new ArrayList<>();
        
         while (!this.io.isEof()) {
             this.packets.add(new Packet(this.io, this));
         }
```